lib/iris/tests/test_analysis.py

```diff
@@ -1155,6 +1155,55 @@ def test_area_weights(self):
         )
 
 
+@tests.skip_data
+class TestLazyAreaWeights:
+    @pytest.mark.parametrize("normalize", [True, False])
+    @pytest.mark.parametrize("chunks", [None, (2, 3, 4)])
+    @pytest.mark.parametrize(
+        "cube_data",
+        [np.ones((4, 3, 4)), da.ones((4, 3, 4), chunks=(1, 3, 4))],
+    )
+    def test_lazy_area_weights(self, cube_data, chunks, normalize):
+        small_cube = iris.tests.stock.simple_3d()[[0, 0, 0, 0], :, :]
+        small_cube.data = cube_data
+        small_cube.coord("latitude").guess_bounds()
+        small_cube.coord("longitude").guess_bounds()
+
+        area_weights = iris.analysis.cartography.area_weights(
+            small_cube,
+            normalize=normalize,
+            compute=False,
+            chunks=chunks,
+        )
+
+        assert isinstance(area_weights, da.Array)
+
+        # Check that chunksizes are as expected
+        if chunks is not None:
+            assert area_weights.chunksize == chunks
+        else:
+            if isinstance(cube_data, da.Array):
+                assert area_weights.chunksize == cube_data.chunksize
+            else:
+                assert area_weights.chunksize == (4, 3, 4)
+
+        # Check that actual weights are as expected
```

Comment:
Please could you note that these are 'known good output'? I have previously panicked when dependencies have changed test outputs and I had no idea why the original numbers were there!

Reply:
Done in https://github.com/SciTools/iris/pull/5658/commits/8a03706d678de744f616a94fd6a0f87b048c97c1, is this okay like this or do I need to add more?